src/App.tsx

```diff
@@ -2,43 +2,81 @@ import { ChangeEvent, useState } from 'react';
 import styles from './App.module.scss';
 import TasksList from './components/TasksList';
 import NewTask from './components/NewTask';
-
-
+import 'semantic-ui-css/semantic.min.css';
 
 function App() {
-    const [tasksArray, setTasksArray] = useState([{ id: 0, item: "Buy Milk" }, { id: 1, item: "Go Run" }, { id: 2, item: "Make Dinner" }, { id: 3, item: "Bake Cake" }]);
-    const [newTaskItem, setNewTaskItem] = useState('')
+    const [tasksArray, setTasksArray] = useState([
+        { id: 0, item: 'Buy Milk' },
+        { id: 1, item: 'Go Run' },
+        { id: 2, item: 'Make Dinner' },
+        { id: 3, item: 'Bake Cake' },
+    ]);
+    const [newTaskItem, setNewTaskItem] = useState('');
+    const [editedTaskItem, setEditedTaskItem] = useState('');
+    const [editTaskId, setEditTaskId] = useState<number | undefined>();
 
     function textTyping(event: ChangeEvent<HTMLInputElement>) {
-        setNewTaskItem(event.target.value)
+        const { name } = event.target;
+        name === 'new-task-text'
+            ? setNewTaskItem(event.target.value)
+            : setEditedTaskItem(event.target.value);
     }
 
-    function clearTyping(){
-        setNewTaskItem("")
+    function clearTyping() {
+        setNewTaskItem('');
     }
 
     function addItem() {
-        setTasksArray([...tasksArray, { id: (tasksArray.length > 0 ) ? tasksArray.slice(-1)[0].id + 1 : 0, item: newTaskItem }])
-        setNewTaskItem("")
-     }
+        setTasksArray([
+            ...tasksArray,
+            { id: tasksArray.length > 0 ? tasksArray.slice(-1)[0].id + 1 : 0, item: newTaskItem },
+        ]);
+        setNewTaskItem('');
+    }
 
     function deleteTask(id: number) {
-        const newTasksArray = tasksArray.filter((item) => item.id !== id)
-        setTasksArray(newTasksArray)
+        const newTasksArray = tasksArray.filter((item) => item.id !== id);
+        setTasksArray(newTasksArray);
```

Comment:
The setter function returned from `React.useState` accepts either a new state or a callback that has the previous state so it is possible to manipulate it before setting a new one. In this case, there won't be any effect but it is a better practice.

```suggestion
        setTasksArray(prev => prev.filter((item) => item.id !== id));
```

Reply:
Done.